source/common/http/http1/legacy_parser_impl.cc

```diff
@@ -0,0 +1,178 @@
+#include "common/http/http1/legacy_parser_impl.h"
+
+#include <http_parser.h>
+
+#include <iostream>
+
+#include "common/common/assert.h"
+#include "common/http/http1/parser.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+
+class LegacyHttpParserImpl::Impl {
+public:
+  Impl(http_parser_type type) {
+    http_parser_init(&parser_, type);
+    parser_.allow_chunked_length = 1;
+  }
+
+  Impl(http_parser_type type, void* data) : Impl(type) {
+    parser_.data = data;
+    settings_ = {
+        [](http_parser* parser) -> int {
+          return static_cast<ParserCallbacks*>(parser->data)->onMessageBegin();
+        },
+        [](http_parser* parser, const char* at, size_t length) -> int {
+          return static_cast<ParserCallbacks*>(parser->data)->onUrl(at, length);
+        },
+        // TODO(dereka) onStatus
```

Comment:
I think we can get rid of these TODO comments (`on_status` and `on_chunk_complete`). We don't use them today.